### PR TITLE
add olmo support

### PR DESCRIPTION
**Pull request by eaidova**
# What does this PR do?
support olmo model. This model is added in transformers 4.40 release. Verfied locally that it work (required update transformers version in optimum)

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by eaidova** [comment]
@echarlaix could you please take a look?

**Review by AlexKoff88 — APPROVED** [review]

File: tests/openvino/test_modeling.py [hunk 1]
```diff
@@ -528,6 +528,7 @@ class OVModelForCausalLMIntegrationTest(unittest.TestCase):
         "mistral",
         "mixtral",
         "mpt",
+        # "olmo", # TODO : enable after optimum update for support transformers 4.40
```
Comment (echarlaix) [inline tests/openvino/test_modeling.py:531]:
```suggestion
        "olmo",
```
patch release is out : https://pypi.org/project/optimum/#history

File: optimum/exporters/openvino/model_configs.py [hunk 2]
```diff
@@ -437,3 +437,9 @@ class OrionOpenVINOConfig(TextDecoderWithPositionIdsOnnxConfig):
     DUMMY_INPUT_GENERATOR_CLASSES = (DummyTextInputGenerator, MistralDummyPastKeyValuesGenerator)
     DUMMY_PKV_GENERATOR_CLASS = MistralDummyPastKeyValuesGenerator
     NORMALIZED_CONFIG_CLASS = NormalizedTextConfig
+
+
+@register_in_tasks_manager("olmo", *["text-generation", "text-generation-with-past"], library_name="transformers")
+class OLMoOpenVINOConfig(TextDecoderWithPositionIdsOnnxConfig):
```
Comment (echarlaix) [inline optimum/exporters/openvino/model_configs.py:443]:
```suggestion
class OlmoOpenVINOConfig(TextDecoderWithPositionIdsOnnxConfig):
```

**Review by echarlaix — APPROVED** [review]

File: tests/openvino/test_modeling.py (hunk 1 above)
Comment (echarlaix) [reply tests/openvino/test_modeling.py:531]: @eaidova let's enable this test before merging

File: optimum/exporters/openvino/model_configs.py [hunk 3]
```diff
@@ -441,6 +441,12 @@ class OrionOpenVINOConfig(TextDecoderWithPositionIdsOnnxConfig):
     NORMALIZED_CONFIG_CLASS = NormalizedTextConfig
 
 
+@register_in_tasks_manager("olmo", *["text-generation", "text-generation-with-past"], library_name="transformers")
+class OlmoOpenVINOConfig(TextDecoderWithPositionIdsOnnxConfig):
+    DEFAULT_ONNX_OPSET = 14
+    NORMALIZED_CONFIG_CLASS = NormalizedTextConfig
+
```
Comment (echarlaix) [inline optimum/exporters/openvino/model_configs.py:448]:
```suggestion

```

File: optimum/exporters/openvino/model_configs.py [hunk 4]
```diff
@@ -441,6 +441,11 @@ class OrionOpenVINOConfig(TextDecoderWithPositionIdsOnnxConfig):
...
```
Comment (echarlaix) [inline optimum/exporters/openvino/model_configs.py:448]:
```suggestion


```

**Pull request merged**